server/services/meilisearch/connector.js

```diff
@@ -258,29 +275,34 @@ module.exports = ({ strapi, adapter, config }) => {
 
       if (!Array.isArray(entries)) entries = [entries]
 
-      const indexUid = config.getIndexNameOfContentType({ contentType })
+      const indexUids = config.getIndexNamesOfContentType({ contentType })
       const documents = await sanitizeEntries({
         contentType,
         entries,
         config,
         adapter,
       })
 
-      const task = await client
-        .index(indexUid)
-        .addDocuments(documents, { primaryKey: '_meilisearch_id' })
+      const tasks = await Promise.all(
+        indexUids.map(async indexUid => {
+          const task = await client
+            .index(indexUid)
+            .addDocuments(documents, { primaryKey: '_meilisearch_id' })
 
-      await store.addIndexedContentType({ contentType })
+          await store.addIndexedContentType({ contentType })
 
-      strapi.log.info(
-        `The task to add ${documents.length} documents to the Meilisearch index "${indexUid}" has been enqueued (Task uid: ${task.taskUid}).`,
+          strapi.log.info(
+            `The task to add ${documents.length} documents to the Meilisearch index "${indexUid}" has been enqueued (Task uid: ${task.taskUid}).`,
+          )
+          return task
+        }),
       )
 
-      return task
+      return tasks.flat()[0]
```

Comment:
Forcefully using only the result of one of the indexing tasks? 🤔 

by the way, do you know if this is used anywhere else?

Reply:
I didn't want to change the result but it's true that the it isn't used anywhere from what I saw -> we just catch the error in `lifecycle` if it fails, I am going to change the result